File: Back-End/src/main/java/fr/ensicaen/pi/gpss/backend/database/dto/direct_debit/DirectDebitDTO.java
package fr.ensicaen.pi.gpss.backend.database.dto.direct_debit;

import com.fasterxml.jackson.annotation.JsonInclude;
import fr.ensicaen.pi.gpss.backend.database.dto.TemplateDTO;
import fr.ensicaen.pi.gpss.backend.database.enumerate.Schedule;
import jakarta.validation.Valid;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Builder;
import lombok.Getter;

@Builder
@Getter
@JsonInclude(JsonInclude.Include.NON_NULL)
@Valid
public class DirectDebitDTO extends TemplateDTO {
    @PositiveOrZero
    private Integer idDirectDebit;
    private String iban;
    private Schedule recurrence;
    private IndividualDTO individual;
    private LegalEntityDTO legalEntity;
}

File: Front-End/src/app/DTO/direct_debit/DirectDebit.dto.ts
import {IndividualDTO} from "./Individual.dto";
import {LegalEntityDTO} from "./LegalEntity.dto";

export interface DirectDebitDTO {
    recurrence: string;
    idDirectDebit: number;
    individual: IndividualDTO | undefined;
    legalEntity: LegalEntityDTO | undefined;
}
